aws_lambda_powertools/utilities/idempotency/idempotency.py

```diff
@@ -91,7 +91,17 @@ def handler(event, context):
         return handler(event, context, **kwargs)
 
     config = config or IdempotencyConfig()
-    config.register_lambda_context(context)
+
+    if hasattr(context, "state"):
+        # Extract lambda_context from DurableContext
+        durable_context = cast("DurableContext", context)
+        config.register_lambda_context(durable_context.lambda_context)
+        # Note: state.operations is accessed via duck typing at runtime
+        is_replay = len(durable_context.state.operations) > 1  # type: ignore[attr-defined]
+    else:
+        # Standard LambdaContext
+        config.register_lambda_context(context)
+        is_replay = False
```

Comment:
I think if we change `hasattr(context, "state")` to `hasattr(context, "state") and hasattr(context, "lambda_context")`. We have more explicit check that ensures we only treat it as a durable context when both required attributes exist.

```suggestion
    # Check for durable context using duck typing
    if hasattr(context, "state") and hasattr(context, "lambda_context"):
        durable_context = cast("DurableContextProtocol", context)
        config.register_lambda_context(durable_context.lambda_context)
        # Note: state.operations is accessed via duck typing at runtime
        is_replay = len(durable_context.state.operations) > 1  # type: ignore[attr-defined]
    else:
        # Standard LambdaContext
        config.register_lambda_context(context)
        is_replay = False
```